src/ApiCore/TransportFactory.php

```diff
@@ -136,35 +77,36 @@ public static function build(array $args)
                         'gRPC extension please see https://cloud.google.com/php/grpc.'
                     );
                 }
-                $stubOpts = [
-                    'force_new' => $args['forceNewChannel']
-                ];
-                // We need to use array_key_exists here because null is a valid value
-                if (!array_key_exists('sslCreds', $args)) {
-                    $stubOpts['credentials'] = self::createSslChannelCredentials();
-                } else {
-                    $stubOpts['credentials'] = $args['sslCreds'];
-                }
+
+                $stubOpts = ['credentials'=> self::createSslChannelCredentials()];
 
                 return new GrpcTransport(
-                    $host,
+                    $serviceAddress,
                     $authWrapper,
-                    $stubOpts,
-                    $args['channel']
+                    $stubOpts
                 );
             case 'rest':
                 self::validateNotNull($args, ['restClientConfigPath']);
 
+                $requestBuilder = new RequestBuilder(
+                    $uri,
+                    $args['restClientConfigPath']
+                );
+
+                try {
+                    $httpHandler = [HttpHandlerFactory::build(), 'async'];
+                } catch (\Exception $ex) {
+                    throw new ValidationException("Failed to create httpHandler", $ex->getCode(), $ex);
+                }
+
                 return new RestTransport(
-                    new RequestBuilder(
-                        $host,
-                        $args['restClientConfigPath']
-                    ),
+                    $requestBuilder,
                     $authWrapper,
-                    $args['httpHandler'] ?: [HttpHandlerFactory::build(), 'async']
```

Comment:
This allowed for supplying a custom function to handle the REST calls. Why is it removed?